### PR TITLE
Unit-tests for model-package-group

### PR DESCRIPTION
**Pull request by ryansteakley**
Description of changes:
Sdk.go coverage: 96.1%
Overall coverage: 64.3% 
Covers all custom code except this portion of [delete requeue](https://github.com/aws-controllers-k8s/sagemaker-controller/blob/f210ea74ef1bfa2c2e40f0a6d7b485fe6cb1249f/pkg/resource/model_package_group/sdk.go#L238).
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/model_package_group/sdk.go [hunk 1]
```diff
@@ -17,8 +17,6 @@ package model_package_group
 
 import (
 	"context"
-	"strings"
```
Comment (surajkota) [inline pkg/resource/model_package_group/sdk.go:20]: revert?, curious why we see these changes only on your PR and why now

File: pkg/resource/model_package_group/testdata/sdkapi/describe/deleting.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (surajkota) [inline pkg/resource/model_package_group/testdata/sdkapi/describe/deleting.json:1]: [delete api](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.delete_model_package_group) doesnt return anything, you can just leave the output_fixture empty in test_suite and that should work

File: pkg/resource/model_package_group/testdata/v1alpha1/update/desired/updated_base.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+apiVersion: sagemaker.services.k8s.aws/v1alpha1
+kind: ModelPackageGroup
+metadata:
+  creationTimestamp: null
+  name: xgboost-model-package-group
+spec:
+  modelPackageGroupDescription: Description for model package group
```
Comment (surajkota) [inline pkg/resource/model_package_group/testdata/v1alpha1/update/desired/updated_base.yaml:7]: nit: whats updated in this file?

**Review by surajkota — CHANGES_REQUESTED** [review]
minor comments. What is the coverage of custom code?

can you add [spec check using gocmp](https://github.com/aws-controllers-k8s/sagemaker-controller/blob/main/pkg/resource/model/manager_test_suite_test.go#L105-L123)?
eventually, we might be able to deprecate the `yamlEqual` function

also, [this test](https://github.com/aws-controllers-k8s/sagemaker-controller/blob/main/pkg/resource/endpoint_config/testdata/test_suite.yaml#L73) might help with some more coverage for update conditions

File: pkg/resource/model_package_group/sdk.go (hunk 1 above)
Comment (ryansteakley) [reply pkg/resource/model_package_group/sdk.go:20]: after adding the go fmt etc to develop unit tests everytime I ran gofmt it does this. I can revert it however

File: pkg/resource/model_package_group/testdata/v1alpha1/create/desired/success_before_create.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+spec:
+  modelPackageGroupName: xgboost-model-package-group
+  modelPackageGroupDescription: "Description for model package group"
+  tags:
+    - key: algorithm
+      value: xgboost
+    - key: environment
+      value: testing
+    - key: customer
+      value: test-user
```
Comment (akartsky) [inline pkg/resource/model_package_group/testdata/v1alpha1/create/desired/success_before_create.yaml:14]: nit : do we need all these tags in unit test 
in every unit test ?

**Review by akartsky — APPROVED** [review]

File: pkg/resource/model_package_group/testdata/sdkapi/describe/deleting.json (hunk 2 above)
Comment (ryansteakley) [reply pkg/resource/model_package_group/testdata/sdkapi/describe/deleting.json:1]: this is a call to` - operation: DescribeModelPackageGroupWithContext` not to delete

File: pkg/resource/model_package_group/testdata/v1alpha1/update/desired/updated_base.yaml (hunk 3 above)
Comment (ryansteakley) [reply pkg/resource/model_package_group/testdata/v1alpha1/update/desired/updated_base.yaml]: changed to try to update description

File: pkg/resource/model_package_group/testdata/v1alpha1/create/desired/success_before_create.yaml (hunk 4 above)
Comment (ryansteakley) [reply pkg/resource/model_package_group/testdata/v1alpha1/create/desired/success_before_create.yaml:14]: Helps with coverage in sdk.go create but I guess not necessary for all tests

**Pull request merged**